### PR TITLE
Allow setting custom filenames for documents

### PR DESCRIPTION
**Pull request by samuelhwilliams**
Updates the REST and client docs to describe the new 'custom filename' feature for sending files by email.

### CODE REVIEW

File: source/documentation/_api_docs.md [hunk 1]
```diff
@@ -361,15 +361,21 @@ Pass the encoded string into an object with a `file` key, and put that in the pe
 }
 ```
 
-##### CSV Files
+#### Set the filename
```
Comment (karlchillmaid) [inline source/documentation/_api_docs.md:364]:
```suggestion
#### filename
```

File: source/documentation/_api_docs.md [hunk 2]
```diff
@@ -361,15 +361,21 @@ Pass the encoded string into an object with a `file` key, and put that in the pe
...
 
-Uploads for CSV files should set the `is_csv` flag as `true` to ensure it’s downloaded as a .csv file. For example:
+You can provide a filename to set when the recipient downloads the file. If this is not provided, a random filename will be generated.
+
+Choosing a sensible filename can help users understand what the file contains, and find it again later.
+
+You do not have to set this, but we strongly recommend it.
+
+The filename must include the correct file extension, such as `.csv` for a CSV file. If you include the wrong file extension, users may not be able to open your file.
```
Comment (karlchillmaid) [inline source/documentation/_api_docs.md:372]:
```suggestion
You should provide a filename when you upload your file.

The filename should tell the recipient what the file contains. A memorable filename can help the recipient to find the file again later.

The filename must end with a file extension. For example, `.csv` for a CSV file. If you include the wrong file extension, recipients may not be able to open your file.

If you do not provide a filename, Notify will generate a random filename for you.
```

Comment (samuelhwilliams) [reply source/documentation/_api_docs.md:372]: > If you do not provide a filename, Notify will generate a random filename for you.

One other bit of behaviour here - if they don't provide a filename (with extension), we also try to _guess_ what kind of file they're sending.

By providing a filename we don't have to guess (with the small chance we guess wrong and either reject the file, or make it harder for the recipient to open).

Don't know how we condense that down to be easily digestible in the docs though 😬

Comment (karlchillmaid) [reply source/documentation/_api_docs.md:372]: >If you do not provide a filename for your file, Notify will:
>
>* generate a random filename
>* try to add the correct file extension
>
>If Notify does not add the correct file extension, recipients may not be able to open your file.

Comment (samuelhwilliams) [reply source/documentation/_api_docs.md:372]: That sounds good on the whole 👍

It's an option to just not mention it (we don't at the moment, and it applies now anyway). But I think tech docs are a place where it's OK (and encouraged) to specific about the technical details.

**Review by CrystalPea — APPROVED** [review]

**Comment by samuelhwilliams** [comment]
@karlchillmaid merging this now because I want to release while I'm still around. But please do finish the review and make any required changes when you're back 🙏

P.S. Pea's changes mean that all the client documentation is now in this repo, so only one place to update :)

**Pull request merged**